Improved: partyRole entity (OFBIZ-5980)

PierreSmits opened the pull request:
added fields 'fromDate' and 'thruDate' to allow for party role expiration

PierreSmits committed:
Improved: partyRole entity (OFBIZ-5980)

added fields 'fromDate' and 'thruDate' to allow for party role expiration

PierreSmits committed:
Improved: party role expiration (OFBIZ-5980)

added: fromDate field to primary key

PierreSmits committed:
Improved: party roles (OFBIZ-5980)

updated data set regarding party roles

JacquesLeRoux reviewed (changes requested):
I don't think we want to remove the comment

<!-- data completed by components loaded later -->

 in HumanresDemoData.xml

PierreSmits commented:
Hi @JacquesLeRoux,

Regularly you advice other contributors to leave the comment on the associated ticket. Please stick to that advice too.

PierreSmits committed:
Improved: party role expiration (OFBIZ-5980)

ensuring that from and thru data are visible in the overview of the party

PierreSmits committed:
Improvement: party role expiration (OFBIZ-5980)

adding from date field to forms associated with the viewroles screen.

JacquesLeRoux commented:
In this case it's because I made a review here.

PierreSmits committed:
Improved: expiration of party roles (OFBIZ-5980)

added/changed:
- in services.xml: updatePartyRole service
- PartyServices.groovy: updatePartyRole
- in services_party.xml: expirePartyRole
- controller.xml: replaced createPartyRole with updatePartyRole, added expirePartyRole
- in PartyForms.xml:  updated grid ViewPartyRoles
- in PartyScreens: updated screen viewroles

PierreSmits committed:
improved: revocation of party roles (OFBIZ-5980)

Cosmetic change.

PierreSmits committed:
Merge branch 'apache:trunk' into ofbiz-5980-PartyRole-expiration

PierreSmits committed:
Improved: seed data (OFBIZ-12353) (#324)

added: employer roletype

PierreSmits committed:
Improved: expiring party roles (OFBIZ-5980)

correcting label for through date field
